### PR TITLE
useLoginWithGoogle hook added

### PR DESCRIPTION
**Pull request by Krriishhhhhh**
Completes : #129 

A hook named useLoginWithGoogle is added in **\packages\client\src\hooks\useLoginWithGoogle.ts** to implement the Signin with Google feature

### CODE REVIEW

File: packages/client/src/hooks/useLoginWithGoogle.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import axios, { AxiosResponse } from "axios";
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const useLoginWithGoogle = () => {
+  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
```
Comment (prasenjeet-symon) [inline packages/client/src/hooks/useLoginWithGoogle.ts:6]: This is not a pure react hook.

**Review by prasenjeet-symon — CHANGES_REQUESTED** [review]
This is a react-query hook not a pure react hook. Plzz see `useLoginWithEmailAndPassword` for inspiration.

**Comment by Krriishhhhhh** [comment]
Can u explain this in a bit detail? I didn't got the issue

**Comment by Krriishhhhhh** [comment]
@prasenjeet-symon kindly review the changes

File: packages/client/src/hooks/authentication.hooks.ts [hunk 2]
```diff
@@ -109,7 +109,7 @@ export function useSignUpWithEmailAndPassword(navigate: NavigateFunction, ctx?:
  * Hook - Authentication
  * Login user with email and password
  */
-export function useLoginWithEmailAndPassword(navigate: NavigateFunction, ctx?: IHookContext): IHookLoginWithEmailAndPassword {
```
Comment (prasenjeet-symon) [inline packages/client/src/hooks/authentication.hooks.ts:112]: Why you deleted this function ?

File: packages/client/src/hooks/authentication.hooks.ts [hunk 3]
```diff
@@ -109,7 +109,7 @@ export function useSignUpWithEmailAndPassword(navigate: NavigateFunction, ctx?:
...
+export function (navigate: NavigateFunction, ctx?: IHookContext): IHookLoginWithEmailAndPassword {
```
Comment (prasenjeet-symon) [inline packages/client/src/hooks/authentication.hooks.ts:112]: Where is function name ?

Comment (prasenjeet-symon) [reply packages/client/src/hooks/authentication.hooks.ts:112]: You should not touch this function, you should take inspiration to build your hook

File: packages/client/src/hooks/useLoginWithGoogle.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        }else(
+            console.error('Authentication Failed')
+        )
+    }catch(error){
+        console.error('Error during authentication:', error);
+    }
+   
+  };
+
+  useEffect(() => {
```
Comment (prasenjeet-symon) [inline packages/client/src/hooks/useLoginWithGoogle.ts:25]: Why use useEffect ? .

File: packages/client/src/hooks/useLoginWithGoogle.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import { useEffect, useState } from "react";
+import {NavigateFunction } from "react-router-dom";
+
+const useLoginWithGoogle = (navigate:NavigateFunction) => {
+  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
+  
+
+  const handleLoginWithGoogle = async (googleToken: string) => {
+    try{
+        const response:AxiosResponse<unknown> = await axios.post("/auth/google_login", { token: googleToken });
```
Comment (prasenjeet-symon) [inline packages/client/src/hooks/useLoginWithGoogle.ts:11]: This should be in a separate file , there is file for Axios related task

**Review by prasenjeet-symon — CHANGES_REQUESTED** [review]
Write react-query hook for GoogleLogin.

File: packages/client/src/hooks/authentication.hooks.ts (hunk 3 above)
Comment (Krriishhhhhh) [reply packages/client/src/hooks/authentication.hooks.ts:112]: Sorry , happened by mistake

File: packages/client/src/hooks/useLoginWithGoogle.ts (hunk 4 above)
Comment (prasenjeet-symon) [reply packages/client/src/hooks/useLoginWithGoogle.ts:25]: Yes , Plzzz see the other hook implementation. You will get the idea .

**Pull request closed without merging**